src/Admin/ORM/MediaAdmin.php

```diff
@@ -49,16 +49,23 @@ protected function configureDatagridFilters(DatagridMapper $filter): void
             ->add('height')
             ->add('contentType');
 
-        $providers = [];
+        $providersChoices = [];
 
-        $providerNames = $this->pool->getProviderNamesByContext($this->getPersistentParameter('context', $this->pool->getDefaultContext()));
-        foreach ($providerNames as $name) {
-            $providers[$name] = $name;
+        $providers = $this->pool->getProvidersByContext($this->getPersistentParameter('context', $this->pool->getDefaultContext()));
+        foreach ($providers as $provider) {
+            $name = $provider->getName();
+            $translatedName = $this->getTranslator()->trans(
+                $name,
+                [],
+                $provider->getProviderMetadata()->getDomain() ?? $this->getTranslationDomain()
```

Comment:
Since someone can add custom providers, it needs a way to add a domain for translations for its new provider. Just like its done here: 

https://github.com/sonata-project/SonataMediaBundle/blob/4.x/src/Resources/views/MediaAdmin/select_provider.html.twig#L38